### PR TITLE
feat(forge): Unbreak `forge` agent

### PR DESCRIPTION
**Pull request by kcze**
### Background

After refactor https://github.com/Significant-Gravitas/AutoGPT/pull/7117 agents created using forge don't work.
- https://github.com/Significant-Gravitas/AutoGPT/issues/7186

### Changes 🏗️

Revert some changes to fix forge agents and enable rudimentary components support for them. LLM calls aren't supported because the providers are in `autogpt`.
- Renamed `forge` `Agent` to `ProtocolAgent` (it's providing agent protocol server)
- Brought back and update `forge/app.py` and `forge/agent/forge_agent.py`
- `ForgeAgent` inherits from `BaseAgent`, supports component execution and runs the same pipelines as `autogpt` `Agent`
- Updated forge version from `0.1.0` to `0.2.0`
- Updated code comments

### PR Quality Scorecard ✨

- [x] Have you used the PR description template? &ensp; `+2 pts`
- [x] Is your pull request atomic, focusing on a single change? &ensp; `+5 pts`
- [x] Have you linked the GitHub issue(s) that this PR addresses? &ensp; `+5 pts`
- [ ] Have you documented your changes clearly and comprehensively? &ensp; `+5 pts`
- [ ] Have you changed or added a feature? &ensp; `-4 pts`
  - [ ] Have you added/updated corresponding documentation? &ensp; `+4 pts`
  - [ ] Have you added/updated corresponding integration tests? &ensp; `+5 pts`
- [ ] Have you changed the behavior of AutoGPT? &ensp; `-5 pts`
  - [ ] Have you also run `agbenchmark` to verify that these changes do not regress performance? &ensp; `+10 pts`

### CODE REVIEW

File: cli.py [hunk 1]
```diff
@@ -149,10 +149,10 @@ def start(agent_name: str, no_setup: bool):
...
             click.echo()
 
-        subprocess.Popen(["./run_benchmark", "serve"], cwd=agent_dir)
-        click.echo("⌛ (Re)starting benchmark server...")
-        wait_until_conn_ready(8080)
-        click.echo()
+        # subprocess.Popen(["./run_benchmark", "serve"], cwd=agent_dir)
+        # click.echo("⌛ (Re)starting benchmark server...")
+        # wait_until_conn_ready(8080)
+        # click.echo()
```
Comment (kcze) [inline cli.py:156]: Doesn't work: `Command not found: agbenchmark`

Comment (ntindle) [reply cli.py:156]: :? we should open an issuee to fix this :(

**Review by ntindle — APPROVED** [review]

File: cli.py (hunk 1 above)
Comment (Pwuts) [reply cli.py:156]: nit:
```diff
+        # FIXME: Doesn't work: Command not found: agbenchmark
```

File: forge/forge/agent/agent.py
Comment (Pwuts) [inline forge/forge/agent/agent.py:1]: -> `protocol_agent.py`
or
-> `forge/agent_protocol/agent.py` if you think that makes sense

File: forge/forge/agent/forge_agent.py [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+        The agent protocol, which is the core of the Forge,
+        works by creating a task and then executing steps for that task.
+        This method is called when the agent is asked to create a task.
+
+        We are hooking into function to add a custom log message.
+        Though you can do anything you want here.
+        """
+        task = await super().create_task(task_request)
+        logger.info(
+            f"📦 Task created: {task.task_id} "
```
Comment (Pwuts) [inline forge/forge/agent/forge_agent.py]:
```suggestion
            f"📦 Task created with ID {task.task_id} and "
```

File: forge/forge/agent/forge_agent.py [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+            result = ActionSuccessResult(outputs=command_result)
+        except AgentTerminated:
+            result = ActionSuccessResult(outputs="Agent terminated or finished")
+        except AgentException as e:
+            result = ActionErrorResult.from_exception(e)
+            logger.warning(f"{tool} raised an error: {e}")
+
+        await self.run_pipeline(AfterExecute.after_execute, result)
+
+        logger.debug("\n".join(self.trace))
```
Comment (Pwuts) [inline forge/forge/agent/forge_agent.py:216]: Typecheck fails here: 
`forge_agent.py:198:30 - error: "command" is possibly unbound (reportPossiblyUnboundVariable)`

**Review by Pwuts — CHANGES_REQUESTED** [review]

**Comment by Pwuts** [comment]
Pretty clean! Nice :)

File: forge/forge/agent/agent.py
Comment (kcze) [reply forge/forge/agent/agent.py:1]: Moved to `forge/agent_protocol/agent.py`

File: forge/forge/agent/forge_agent.py (hunk 3 above)
Comment (kcze) [reply forge/forge/agent/forge_agent.py:216]: fixed
```py
command: Optional[Command] = None
for c in reversed(self.commands):
    if tool.name in c.names:
        command = c

if command is None:
    raise AgentException(f"Command {tool.name} not found")
```

File: cli.py (hunk 1 above)
Comment (kcze) [reply cli.py:156]: Added comment and Linear issue

File: forge/forge/agent/agent_test.py
Comment (Pwuts) [inline forge/forge/agent/agent_test.py:1]: this file should be moved to where agent.py is now

File: forge/forge/agent_protocol/api_router.py [hunk 4]
```diff
@@ -24,7 +24,7 @@
 )
 
 if TYPE_CHECKING:
-    from forge.agent.agent import ProtocolAgent
+    from forge.agent_protocol.agent import ProtocolAgent
```
Comment (Pwuts) [inline forge/forge/agent_protocol/api_router.py]: nit:
```suggestion
    from .agent import ProtocolAgent
```

File: forge/forge/agent/agent_test.py [hunk 5]
```diff
@@ -13,7 +13,7 @@
 from forge.file_storage.base import FileStorageConfiguration
 from forge.file_storage.local import LocalFileStorage
 
-from .agent import ProtocolAgent
+from ..agent_protocol.agent import ProtocolAgent
```
Comment (Pwuts) [inline forge/forge/agent/agent_test.py]:
```suggestion
from forge.agent_protocol.agent import ProtocolAgent
```

**Review by ntindle — APPROVED** [review]

**Comment by ntindle** [comment]
@kcze can we work on getting this ready to merge asap? Lots of complaints lol

File: forge/forge/agent/agent_test.py (hunk 5 above)
Comment (kcze) [reply forge/forge/agent/agent_test.py]: I moved the `agent_test.py` and updated imports

**Comment by Torantulino** [comment]
Fixes #7154

**Pull request merged**